clang/lib/CIR/Lowering/DirectToLLVM/LowerToLLVM.cpp

```diff
@@ -795,9 +795,24 @@ class CIRIfLowering : public mlir::OpConversionPattern<mlir::cir::IfOp> {
     }
 
     rewriter.setInsertionPointToEnd(currentBlock);
-    auto trunc = rewriter.create<mlir::LLVM::TruncOp>(loc, rewriter.getI1Type(),
-                                                      adaptor.getCondition());
-    rewriter.create<mlir::LLVM::CondBrOp>(loc, trunc.getRes(), thenBeforeBody,
+
+    // Redundant zext + trunc sequence prevents lowering of llvm.expect in
```

Comment:
This should be a `TODO` or `FIXME`. Thanks for improving this, but I still a bit unclear what is the actionable thing here for a new developer looking at this code. Should they change this once some PR lands? Should something be improved before this can be fixed?

Reply:
Updated